src/arrayMethodSort.js

```diff
@@ -5,8 +5,83 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    let copyCompare = compareFunction;
```

Comment:
to avoid such a complicated solution let's make next steps:
1.  add a default parameter for compareFunction to compare as strings
```(a, b) => String(a) > String(b)```

2. use loop inside another loop to work with elements
```
    for (let i = 0; i < this.length; i++) {
      for (let j = 0; j < this.length - i - 1; j++) {
```

3. inside inner loop create a condition using `compareFunction`
```
if (compareFunction(this[j], this[j + 1]) > 0) {
```

4. if we pass a condition switch values
```
const temp = this[j + 1];

this[j + 1] = this[j];
this[j] = temp;
```

5. after loops just return `this`

Reply:
Thank you, it turned out much better. But I don't understand why the function "compareFunction(this[j], this[j + 1]) > 0" returns a number instead of a boolean value.